lib/Driver/Driver.cpp

```diff
@@ -1751,7 +1756,10 @@ static StringRef getOutputFilename(Compilation &C,
   // FIXME: Treat GeneratePCHJobAction as non-top-level (to get tempfile and not
   // use the -o arg) even though, based on ownership considerations within the
   // driver, it is stored as a "top level" JobAction.
-  if (isa<GeneratePCHJobAction>(JA)) {
+  if (auto *PCHAct = dyn_cast<GeneratePCHJobAction>(JA)) {
+    // For a persistent PCH use the PCH output directory as the -o arg.
+    if (PCHAct->isPersistentPCH())
+      return PCHAct->getPersistentPCHDir();
```

Comment:
Oh, this also seems a little weird: isn't this function supposed to be returning a filename for -o, not a directory?